libyul/YulName.h

```diff
@@ -0,0 +1,30 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Definition of the YulName, an identifier for Handles.
+ */
+
+#pragma once
+
+namespace solidity::yul
+{
+
+class YulString;
+using YulName = YulString;
```

Comment:
Unless you're planning to add something more here, like utilities (or maybe make `YulName` into a class in the future), it would be enough to put it in `AST.h`. We'll need it pretty much only in Yul AST. In fact, if there is some situation where it does not seem appropriate to include the whole `AST.h`, it could be a sign that `YulString` there is not really used for identifiers.